File: 0658-find-k-closest-elements/0658-find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        int low = 0;
        int high = arr.length - 1;
        while (high - low >= k) {
            if(Math.abs(arr[low] - x) > Math.abs(arr[high] - x)) {
                low++;
            } else {
                high--;
            }
        }
        List<Integer> result = new ArrayList<>(k);
        for (int i = low; i <= high; i++) {
            result.add(arr[i]);
        }
        return result;
    }
}

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        // Comparator to check the conditions
        auto cmp = [&] (int a, int b) {
            int da = abs(a-x);
            int db = abs(b-x);
            if (da != db) return da < db;
            return a < b;
        };
        // Making the Heap
        priority_queue<int, vector<int>, decltype(cmp)> maxh(cmp);
        // Heap containerisation
        for (int val : arr) {
            maxh.push(val);
            if (maxh.size() > k) maxh.pop();
        }
        // Return the answer
        vector<int> ans;
        while (!maxh.empty()) {
            ans.push_back(maxh.top());
            maxh.pop();
        }
        sort(ans.begin(), ans.end());
        return ans;
    }
};